./bootstrap-vcpkg.sh on mac selects wrong compiler

**Describe the bug**
When I run `./bootstrap-vcpkg.sh` I get the following error:
```
➜  vcpkg git:(fix-meson) ✗ ./bootstrap-vcpkg.sh                                                         
Downloading vcpkg tool sources
Building vcpkg-tool...
-- The CXX compiler identification is GNU 11.2.0
-- Checking whether CXX compiler has -isysroot
-- Checking whether CXX compiler has -isysroot - yes
-- Checking whether CXX compiler supports OSX deployment target flag
-- Checking whether CXX compiler supports OSX deployment target flag - yes
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /usr/local/bin/g++-11
-- Check for working CXX compiler: /usr/local/bin/g++-11 - broken
CMake Error at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/tools/cmake-3.20.2-osx/cmake-3.20.2-macos-universal/CMake.app/Contents/share/cmake-3.20/Modules/CMakeTestCXXCompiler.cmake:59 (message):
  The C++ compiler

    "/usr/local/bin/g++-11"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/_vcpkg/build/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/tools/ninja-1.10.2-osx/ninja cmTC_3da22 && [1/2] Building CXX object CMakeFiles/cmTC_3da22.dir/testCXXCompiler.cxx.o
    FAILED: CMakeFiles/cmTC_3da22.dir/testCXXCompiler.cxx.o 
    /usr/local/bin/g++-11   -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -o CMakeFiles/cmTC_3da22.dir/testCXXCompiler.cxx.o -c testCXXCompiler.cxx
    g++-11: error: this compiler does not support arm64
    ninja: build stopped: subcommand failed.

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:33 (project)


-- Configuring incomplete, errors occurred!
See also "/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/_vcpkg/build/CMakeFiles/CMakeOutput.log".
See also "/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/_vcpkg/build/CMakeFiles/CMakeError.log".
```
I can workaround this by passing the option `-allowAppleClang` to the script, but it should autoselect clang on arm macs or on macOS >= 10.15(?)

**Environment**
- OS: macOS 11 with Apple M1

**Expected behavior**
It should simply compile vcpkg

--- a/scripts/bootstrap.sh
+++ b/scripts/bootstrap.sh
@@ -268,7 +268,7 @@ else
     fetchTool "ninja" "$UNAME" ninjaExe || exit 1
 fi
 if [ "$os" = "osx" ]; then
-    if [ "$vcpkgAllowAppleClang" = "true" ] ; then
+    if [ "$vcpkgAllowAppleClang" = "true" ] || [[ $(sw_vers -productVersion | awk -F '.' '{print $1}') -ge 11 ]]; then
         CXX=clang++
     else
         selectCXX

